tests/utils.py

```diff
@@ -0,0 +1,52 @@
+import wandb
+from pydantic import BaseModel
+from wandb.sdk.artifacts.artifact_state import ArtifactState
+
+
+def copy_pydantic_json(model: BaseModel) -> BaseModel:
+    """Copy a Pydantic model through round-trip JSON serialization."""
+    return model.__class__.model_validate_json(model.model_dump_json())
+
+
+class FakeArtifactLoader:
```

Comment:
nit: move out of 'utils.py' to something more descriptive?

Reply:
Ill rename this file in the follow up/